packages/vm/src/testing/querier.rs

```diff
@@ -0,0 +1,231 @@
+use serde::{de::DeserializeOwned, Serialize};
+
+use cosmwasm_std::testing::{MockQuerier as StdMockQuerier, MockQuerierCustomHandlerResult};
+use cosmwasm_std::{
+    to_binary, to_vec, Binary, Coin, Empty, HumanAddr, Querier as _, QueryRequest, StdResult,
+    SystemError, SystemResult,
+};
+
+use crate::{FfiError, FfiResult, GasInfo, Querier};
+
+const GAS_COST_QUERY_FLAT: u64 = 100_000;
+/// Gas per request byte
+const GAS_COST_QUERY_REQUEST_MULTIPLIER: u64 = 0;
+/// Gas per reponse byte
+const GAS_COST_QUERY_RESPONSE_MULTIPLIER: u64 = 100;
+
+/// MockQuerier holds an immutable table of bank balances
+/// TODO: also allow querying contracts
+pub struct MockQuerier<C: DeserializeOwned = Empty> {
+    querier: StdMockQuerier<C>,
+}
+
+impl<C: DeserializeOwned> MockQuerier<C> {
+    pub fn new(balances: &[(&HumanAddr, &[Coin])]) -> Self {
+        MockQuerier {
+            querier: StdMockQuerier::new(balances),
+        }
+    }
+
+    // set a new balance for the given address and return the old balance
+    pub fn update_balance<U: Into<HumanAddr>>(
+        &mut self,
+        addr: U,
+        balance: Vec<Coin>,
+    ) -> Option<Vec<Coin>> {
+        self.querier.update_balance(addr, balance)
+    }
+
+    #[cfg(feature = "staking")]
+    pub fn update_staking(
+        &mut self,
+        denom: &str,
+        validators: &[cosmwasm_std::Validator],
+        delegations: &[cosmwasm_std::FullDelegation],
+    ) {
+        self.querier.update_staking(denom, validators, delegations);
+    }
+
+    pub fn with_custom_handler<CH: 'static>(mut self, handler: CH) -> Self
+    where
+        CH: Fn(&C) -> MockQuerierCustomHandlerResult,
+    {
+        self.querier = self.querier.with_custom_handler(handler);
+        self
+    }
+}
+
+impl<C: DeserializeOwned> Querier for MockQuerier<C> {
+    fn query_raw(
+        &self,
+        bin_request: &[u8],
+        gas_limit: u64,
+    ) -> FfiResult<SystemResult<StdResult<Binary>>> {
+        let response = self.querier.raw_query(bin_request);
```

Comment:
Huh, so the Vm::Querier uses `query_raw` which takes a gas price, and the Std::Querier uses `raw_query` which takes no gas.

I assume this is intentional, but a bit odd to see the different names

Reply:
The fact the the vm MockQuerier uses std's MockQuerier is a bit dirty but works for now. A solid solution would copy the logic of std's MockQuerier and add gas prices to the implementation.